pyaqueduct/experiment.py

```diff
@@ -79,14 +82,22 @@ def tags(self) -> List[str]:
         return self._client.get_experiment(self.id).tags
 
     @validate_call
-    def add_tag(self, tag: str = Field(..., max_length=_MAX_TAG_LENGTH)) -> None:
-        """Add new tag to experiment."""
-        if not is_valid_tag(tag):
-            raise ValueError(
-                f"Tag {tag} should only contain alphanumeric characters, underscores or hyphens"
-            )
-
-        self._client.add_tag_to_experiment(experiment_uuid=self.id, tag=tag)
+    def add_tags(self, tags: List[TagString]) -> None:
+        """Add new tags to experiment.
+
+        Args:
+            tags: List of tags to be added to the experiment.
+
+        """
+
+        for tag in tags:
+            if not is_valid_tag(tag):
+                raise ValueError(
+                    f"Tag {tag} should only contain alphanumeric characters, "
+                    "underscores or hyphens."
+                )
```

Comment:
In case of multiple tags passed with invalid values, this will raise exception for just first invalid tag. I guess this can be replaced with:

```python
invalid_tags = [tag for tag in tags if not is_valid_tag(tag)]
if invalid_tags:
    raise ValueError(
        f"Tags {invalid_tags} should only contain alphanumeric characters, "
        "underscores or hyphens."
    )
```

Reply:
Thanks. This is more helpful to the user, I apply this change.